docs/source/result_format.rst

```diff
@@ -0,0 +1,425 @@
++---------------------------------------+----------------+----------------+----------------+----------------+
+|    observed_value                     |no              |yes             |yes             |yes             |
++---------------------------------------+----------------+----------------+----------------+----------------+
+|    details (e.g. statistical details) |no              |no              |yes             |yes             |
++---------------------------------------+----------------+----------------+----------------+----------------+
+
+
++---------------------------------------+--------------------------------------------------------------+
+| `result_obj_format` Setting           | Example use case                                             |
++=======================================+==============================================================+
+|    BOOLEAN_ONLY                       | Automatic validation. No result_obj is returned.             |
++---------------------------------------+--------------------------------------------------------------+
+|    BASIC                              | Exploratory analysis in a notebook.                          |
++---------------------------------------+--------------------------------------------------------------+
+|    SUMMARY                            | Detailed exploratory work with follow-on investigation.      |
++---------------------------------------+--------------------------------------------------------------+
+|    COMPLETE                           | Debugging pipelines or developing detailed regression tests. |
++---------------------------------------+--------------------------------------------------------------+
+
+
+result_obj_format examples
+------------------------------------------------------------------------------
+
+.. code-block:: bash
+
+    >> print(list(my_df.my_var))
+    ['A', 'B', 'B', 'C', 'C', 'C', 'D', 'D', 'D', 'D', 'E', 'E', 'E', 'E', 'E', 'F', 'F', 'F', 'F', 'F', 'F', 'G', 'G', 'G', 'G', 'G', 'G', 'G', 'H', 'H', 'H', 'H', 'H', 'H', 'H', 'H']
+
+    >> my_df.expect_column_values_to_be_in_set(
+        "my_var",
+        ["B", "C", "D", "F", "G", "H"],
+        result_format={'result_obj_format': 'BOOLEAN_ONLY'}
+    )
+    {
+        'success': False
+    }
+
+    >> my_df.expect_column_values_to_be_in_set(
+        "my_var",
+        ["B", "C", "D", "F", "G", "H"],
+        result_format={'result_obj_format': 'BASIC'}
+    )
+    {
+        'success': False,
+        'result_obj': {
+            'unexpected_count': 6,
+            'unexpected_percent': 0.16666666666666666,
+            'unexpected_percent_nonmissing': 0.16666666666666666,
+            'partial_unexpected_list': ['A', 'E', 'E', 'E', 'E', 'E']
+        }
+    }
+
+    >> expect_column_values_to_match_regex(
+        "my_column",
+        "[A-Z][a-z]+",
+        result_format={'result_obj_format': 'SUMMARY'}
```

Comment:
Can we limit the unexpected count to 3 or something?

Reply:
Just to show an example of that behavior

Reply:
Done